### ISSUE TITLE
How to set a different OpenAPI schema name for automatically generated body names

### ISSUE THREAD

**Issue by lnicalo** [tiangolo/fastapi#1494]
### First check

* [x] I used the GitHub search to find a similar issue and didn't find it.
* [x] I searched the FastAPI documentation, with the integrated search.
* [x] I already searched in Google "How to X in FastAPI" and didn't find any information.

### Description

I have this API method:

```
@app.post("/api/v0/parse_tables/pdf/{pdf_type}/{parser}/{orient}",
          responses={**responses},
          response_model=List[ParsedTable],
          tags=['pdf'],
          summary='Parse tables from pdf',
          response_description="List of parsed tables",
          operation_id='parse_tables_pdf')
async def pdf_parse_tables(
    pdf_type: PdfType,
    orient: JsonTableOrient,
    parser: table_parsers.TableParsers,
    input_files: List[UploadFile] = File(..., description='Input files', title='Input files - title')
):
```
In the swagger, the schema for ` input_files: List[UploadFile]` shows this ugly name `Body_pdf_parse_tables_api_v0_parse_tables_pdf__pdf_type___parser___orient__post`:

![image](https://user-images.githubusercontent.com/9167447/83165893-69829100-a10e-11ea-9442-6003738a6683.png)

How could I set a different name? 

Thanks.

Luis

**Comment by lnicalo** [comment]
@tiangolo Could anyone answer this question? Are more clarifications needed?

**Comment by phy25** [comment]
This is where the name is generated:

https://github.com/tiangolo/fastapi/blob/5984233223777313e95a92d534ab1db711bd606a/fastapi/dependencies/utils.py#L725

Not sure how to work around this except overriding that specific openapi json key.

**Comment by tiangolo** [comment]
Yep, currently, you can alter the name of the generated OpenAPI operation ID with this: https://fastapi.tiangolo.com/advanced/generate-clients/

But that still leaves the whole generated name. You could also modify the resulting OpenAPI: https://fastapi.tiangolo.com/advanced/extending-openapi/

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.
